pkgs/applications/audio/ptcollab/default.nix

```diff
@@ -0,0 +1,34 @@
+{ mkDerivation
+, stdenv
+, fetchFromGitHub
+, qmake
+, qtbase
+, qtmultimedia
+, libvorbis
+}:
+
+mkDerivation rec {
+  pname = "ptcollab";
+  version = "unstable-2020-10-20";
+
+  src = fetchFromGitHub {
+    owner = "yuxshao";
+    repo = "ptcollab";
+    rev = "4827b3e895909f02980140efd30652afe257da8e";
+    sha256 = "1zn76r8dr9pakh5lzqw2hcgc290f20dywy9lnc0fylz342y6147c";
+  };
+
+  nativeBuildInputs = [ qmake ];
+
+  buildInputs = [ qtbase qtmultimedia libvorbis ];
+
+  meta = with stdenv.lib; {
+    description = "Experimental pxtone editor where you can collaborate with friends";
+    homepage = "https://yuxshao.github.io/ptcollab/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ OPNA2608 ];
+    platforms = platforms.all;
+    # Requires Qt5.15
+    broken = stdenv.hostPlatform.isDarwin;
+  };
+}
```

Comment:
(just a pet peeve, [in nixpkgs files should end with a newline](https://github.com/NixOS/nixpkgs/blob/303e7d6ef419230eeadcca63f42c32ba96536284/.editorconfig#L7))

Reply:
Fixed… I think? The line already ended in a newline when I checked with `xxd`, I'm assuming you meant an extra empty line?

Reply:
`insert_final_newline` means that we enforce a linefeed on the final line:
```
  };␊
}␊
```

and reject this:
```
  };␊
}
```

Currently this adding an _extra_ linefeed:
```
  };␊
}␊
␊
```
Some editors will automatically remove this anyway and we don't require it.

Reply:
I get that. But I'm confused what exactly he's seeing different there.

I use `vim_configure` with the `editorconfig-vim` plugin, and the linefeed on the last line *seemed* there. Both when checking from within `vim` with `:set list` and when checking after writing the file to disk with `xxd`…

![grafik](https://user-images.githubusercontent.com/23431373/97083429-3d613280-1610-11eb-9a55-49e3bd451d9e.png)
![grafik](https://user-images.githubusercontent.com/23431373/97083435-4e11a880-1610-11eb-84a5-7aff61bb3daf.png)

I've recommitted, this time without the *extra* extra newline again. Is this still a problem you're seeing on your end?

Reply:
It was a problem on my end, sorry for the noise.

It turns out, after making the switch to vim many years ago, I misinterpreted how [it renders the newline character at the end](https://stackoverflow.com/questions/15639511/vim-show-newline-at-the-end-of-file), and, like the person in that stackoverflow question, I started putting double newlines at the end of most files I created, imitating what I did with other less unix-y editors. That includes 33 of my 49 nix packages that I now have to fix :facepalm: :facepalm: :facepalm:.

Fortunately when reviewing I usually look at the handy "no newline at the end of file" in github, so this should be the first time I actually tell someone else to do the wrong thing...